wasmd/templates/initialize/init.go

```diff
@@ -0,0 +1,126 @@
+package initialize
+
+import (
+	"embed"
+	"path/filepath"
+	"regexp"
+
+	"github.com/gobuffalo/genny/v2"
+	"github.com/gobuffalo/plush/v4"
+	"github.com/ignite/cli/ignite/pkg/placeholder"
+	"github.com/ignite/cli/ignite/pkg/xgenny"
+	"github.com/ignite/cli/ignite/templates/module"
+)
+
+const (
+	PathAppModule = "app"
+	PathAppGo     = "app/app.go"
+)
+
+type PlaceHolder string
+type Change string
+type AppDynamicChanges map[PlaceHolder]Change
+
+type After string
+type Before string
+type AppDynamicReplaces map[Before]After
+
+// files/**/* maybe don't needed
+
+//go:embed files/* files/**/*
+var files embed.FS
+
+//go:embed placeholders/*
+var placeholders embed.FS
+
+// NewGenerator returns the generator to scaffold code to import wasm module inside an app.
+func NewGenerator(opts *InitOptions) (*genny.Generator, error) {
+	g := genny.New()
+
+	filePathVersion := "files/" + opts.Version
+
+	filesTemplate := xgenny.NewEmbedWalker(
+		files,
+		filePathVersion,
+		opts.AppPath+"/"+module.PathAppModule,
+	)
+	if err := g.Box(filesTemplate); err != nil {
+		return g, err
+	}
+
+	ctx := plush.NewContext()
```

Comment:
It might be a good idea to be able to set an optional `context.Context` in `InitOptions`. By default the `InitOptions.context` could be initialized with `context.Background()`. If so then this could be rewritten as:

```suggestion
	ctx := plush.NewContextWithContext(opts.context)
```

Reply:
I have added context to the parent function and I thought that might be better. Let me know what you think about it.

Reply:
Nice, even better 👍